src/qibo/transpiler/router.py

```diff
@@ -651,6 +652,8 @@ def __init__(
         self.circuit = None
         self._memory_map = None
         self._final_measurements = None
+        self._added_swaps = 0
```

Comment:
This variable name can be a bit misleading. The transpiler has the method added_swaps() that returns self.circuit._swaps which is the total number of swaps added during the procedure. I would suggest to call this variable "temporary_added_swaps", "cycle_added_swaps" or something like that to avoid confusion with the total number of swaps.